libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -598,22 +597,84 @@ string YulUtilFunctions::clearStorageRangeFunction(Type const& _type)
 {
 	string functionName = "clear_storage_range_" + _type.identifier();
 
-	solUnimplementedAssert(_type.isValueType(), "...");
+	solAssert(_type.storageBytes() >= 32, "Expected smaller value for storage bytes");
 
 	return m_functionCollector->createFunction(functionName, [&]() {
 		return Whiskers(R"(
 			function <functionName>(start, end) {
-				for {} lt(start, end) { start := add(start, 1) }
+				for {} lt(start, end) { start := add(start, <increment>) }
 				{
-					sstore(start, 0)
+					<setToZero>(start, 0)
 				}
 			}
 		)")
 		("functionName", functionName)
+		("setToZero", storageSetToZeroFunction(_type))
+		("increment", _type.storageSize().str())
 		.render();
 	});
 }
 
+string YulUtilFunctions::clearStorageArrayFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+
+	if (_type.baseType()->storageBytes() < 32)
+	{
+		solAssert(_type.baseType()->isValueType(), "Invalid storage size for non-value type.");
+		solAssert(_type.baseType()->storageSize() <= 1, "Invalid storage size for type.");
+	}
+
+	if (_type.baseType()->isValueType())
+		solAssert(_type.baseType()->storageSize() <= 1, "Invalid size for value type.");
+
+	string functionName = "clear_storage_array_" + _type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		if (_type.isDynamicallySized())
+			return Whiskers(R"(
+				function <functionName>(slot) {
+					<clearDynamicArray>(slot)
+				}
+			)")
+			("functionName", functionName)
+			("clearDynamicArray", clearDynamicStorageArrayFunction(_type))
+			.render();
+		else
+			return Whiskers(R"(
+				function <functionName>(slot) {
+					<clearRange>(slot, add(slot, <lenToSize>(<len>)))
+				}
+			)")
+			("functionName", functionName)
+			(
```

Comment:
```
			(
				"clearRange",
				_type.baseType()->storageBytes() < 32 ?
				clearStorageRangeFunction(*TypeProvider::uint256()) :
				clearStorageRangeFunction(*_type.baseType())
			)
```

Reply:
Oh and actually - shouldn't the ternary operator be put inside the argument of `clearStorageRangeFunction`?